### PR TITLE
Followers Activity implementation

### PR DESCRIPTION
**Pull request by dfernandezm**
- Initial steps for Followers Activity implementation
- Like activity single/multiple view
- Started Follow activity single/multiple with basic generification
- The rest are pending

### CODE REVIEW

File: app/src/main/java/com/graffitab/ui/adapters/followersactivity/GenericFollowersActivityRecyclerViewAdapter.java [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+            default:
+                throw new RuntimeException("The view type " + type + " is not supported");
+        }
+    }
+
+
+    @Override
+    public RecyclerView.ViewHolder onCreateItemViewHolder(ViewGroup parent, int viewType) {
+        switch (viewType) {
+            case VIEW_TYPE_LIKE_SINGLE: {
```
Comment (8masterofpuppets8) [inline app/src/main/java/com/graffitab/ui/adapters/followersactivity/GenericFollowersActivityRecyclerViewAdapter.java]: You can probably group these together under the same `case` statement and do an `if..else` on the `viewType`

File: app/src/main/res/layout/row_activity_follow_single.xml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+        android:layout_gravity="center_horizontal"
+        android:background="#fff">
+
+        <LinearLayout
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:background="?attr/selectableItemBackground"
+            android:orientation="horizontal">
+
+            <include layout="@layout/row_notification_include_left_user" />
```
Comment (8masterofpuppets8) [inline app/src/main/res/layout/row_activity_follow_single.xml]: This should probably be row_activity_include_left_user as it's already created.

File: app/src/main/res/layout/row_activity_follow_single.xml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+            <include layout="@layout/row_notification_include_middle" />
```
Comment (8masterofpuppets8) [inline app/src/main/res/layout/row_activity_follow_single.xml]: This too but row_activity_include_middle

File: app/src/main/res/layout/row_activity_include_middle.xml [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    <TextView
+        android:id="@+id/actionLbl"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:text="Mike Lngmr likes your graffiti"
+        android:textColor="@color/colorPrimary"
+        android:textSize="@dimen/row_notification_middle_title_text_size" />
+
+    <!--Add dinamically to this view-->
+    <LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (8masterofpuppets8) [inline app/src/main/res/layout/row_activity_include_middle.xml]: Can't see where this is used, is it a placeholder?

File: app/src/main/java/com/graffitab/ui/adapters/followersactivity/adapters/FollowersActivityItemsAdapter.java [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+import com.graffitab.R;
+import com.squareup.picasso.Picasso;
+
+import java.util.List;
+
+/**
+ * Created by david on 19/03/2017.
+ */
+
+public class FollowersActivityItemsAdapter<T> extends RecyclerView.Adapter<FollowersActivityItemsAdapter.PhotoDetailViewHolder> {
```
Comment (8masterofpuppets8) [inline app/src/main/java/com/graffitab/ui/adapters/followersactivity/adapters/FollowersActivityItemsAdapter.java]: For this adapter you can try to reuse the GridStreamableViewHolder which already handles thumbnail loading and can be added to any adapter. You would just have to set the type of streamables to display to be GRID. Will save a bit of code and an inner class. Don't worry about clicking on single images - we don't support that, you can only click on the whole row and a separate screen will appear with all streamables ;)

**Comment by 8masterofpuppets8** [comment]
So I think this is basically done. I would just duplicate the layouts from the notifications and remove the unread indicator. I can do some minor UI restylings and we can move to the Facebook friends activity.

**Comment by dfernandezm** [comment]
Ready to rebase and merge. Try to recreate the scrolling issue, no longer able to since I fix a different, initially unrelated thing in the CreateStreamableViewHolder.

**Pull request merged**